### DISCUSSION TITLE
Is the @deprecated directive supported?

### DISCUSSION THREAD

**Discussion by euriasb** [comigor/artemis#366] [Q&A]
Some fields in my schema were recently deprecated and the @deprecated directive was used for the first time, after updating the schema the code generation failed, I manually removed all of the @deprecated tags from the local schema and the generation worked this time, so it seems pretty clear that the directive is not supported, so: 1. is it in fact not supported? 2. is there a workaround?

Thanks

**Comment by comigor** [comment]
Hey there @euriasb! It should be, there are for example [some tests](https://github.com/comigor/artemis/tree/beta/test/query_generator/deprecated) with the annotations.
Can you give an example of the usage on your schema?

**Reply by euriasb** [reply]
I think I found the reason, the library doesn't seem to support `@deprecated` without the explicit `reason` parameter, all the usages in my schema are in this form:

```
type ExecutionRealizedProfits {
  ...
  profitability: Decimal @deprecated
  absoluteProfit: Decimal @deprecated
  totalValue: Decimal @deprecated
  allocation: Decimal @deprecated
  settleCurrency: Currency @deprecated
}
```

I manually added a reason parameter and now it works.

So, what can I do about this? I cannot manually change the schema every time I update it.

**Reply by vasilich6107** [reply]
I'll be able to address this issue next week

**Reply by euriasb** [reply]
Thank you!

**Comment by vasilich6107** [comment]
@euriasb 
I created PR https://github.com/comigor/artemis/pull/367 extended tests and `@deprecated` works fine

If you still experiencing any issues create an issue and fill in the template so we can reproduce the issue

**Comment by vasilich6107** [comment]
@euriasb any feedback?

**Comment by euriasb** [comment]
I'm using v6.18, since 7 is still in beta, so the issue was in v6.18. 
However, I removed all uses of deprecated fields from my code, so this is no longer an issue.
Side question, is v7 stable enough to use in production? I've been meaning to upgrade but I don't feel comfortable using it in production code yet.

**Reply by mdrideout** [reply]
Yes, 7 is the preferred branch. I've been using it for 6 months in production without any problems.

**Reply by euriasb** [reply]
Thanks, I will upgrade soon then. 
Why not promote it to main if it's the preferred branch though?

**Reply by vasilich6107** [reply]
> Thanks, I will upgrade soon then. 
> 
> Why not promote it to main if it's the preferred branch though?

This question goes to @comigor

**Reply by vasilich6107** [reply]
> I'm using v6.18, since 7 is still in beta, so the issue was in v6.18. 
> 
> However, I removed all uses of deprecated fields from my code, so this is no longer an issue.
> 
> Side question, is v7 stable enough to use in production? I've been meaning to upgrade but I don't feel comfortable using it in production code yet.

Do not use 6 
there were tons of bugs fixed in 7.*

**Reply by euriasb** [reply]
Good to know, I'll upgrade in my next release

**Reply by comigor** [reply]
> Thanks, I will upgrade soon then. Why not promote it to main if it's the preferred branch though?

Yeah, I've been willing to do that but there are some known issues, like fragments still with some bugs, and I guess that's why we still haven't promoted it.

But, I mean, v7 is actually more stable than v6, regardless of everything... I guess I should bump it, right?
